python-checks/src/test/resources/checks/hotspots/graphQLIntrospection.py

```diff
@@ -296,9 +296,51 @@ def graphql_server_validation_rules_compliant(schema, some_rule, introspection_r
         validation_rules=[SafeRule, some_rule]
     )
 
-    GraphQLView.as_view( 
+    GraphQLView.as_view(
         name="introspection",
         schema=schema,
         graphiql=True,
         validation_rules=[introspection_rule]
     )
+
+
+def flask_graphql_middleware_list_as_variable(middleware, schema, CustomBackend, format_custom_error):
+    from graphql_server.flask import GraphQLView
+    gql_middlew = [
+        middleware.CostProtectionMiddleware(),
+        middleware.DepthProtectionMiddleware(),
+        middleware.IntrospectionMiddleware(),
+        middleware.processMiddleware(),
+    ]
+
+    igql_middlew = [
+        middleware.IGQLProtectionMiddleware()
+    ]
+
+    GraphQLView.as_view( # OK
+        'graphql',
+        schema=schema,
+        middleware=gql_middlew,
+        backend=CustomBackend(),
+        batch=True
+    )
+
+    GraphQLView.as_view( # Noncompliant
+        'graphiql',
+        schema=schema,
+        backend=CustomBackend(),
+        graphiql=True,
+        middleware = igql_middlew,
+        format_error=format_custom_error
+    )
+
+    from a_module import some_function
+    some_var = some_function()
+    GraphQLView.as_view( # OK
+        'graphiql',
+        schema=schema,
+        backend=CustomBackend(),
+        graphiql=True,
+        middleware=some_var,
+        format_error=format_custom_error
```

Comment:
I think this could be added to the other functions. This way we can keep the same grouping with compliant/noncompliant and stay a bit more consistant.

Reply:
Yes, I was thinking of doing this. I guess I just wanted the diff of this PR to be clean regarding the test file. I will make the change.

Reply:
I reorganized the test file in 985c67818094d9de1507b0021e12c89900ca2790.